plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1264,6 +1333,127 @@ private List<ColumnMetadata> getColumnMetadatas(Table table)
+
+        ConnectorTableMetadata tableMetadata = getTableMetadata(session, handle.getSchemaTableName());
+        Set<String> allColumnNames = tableMetadata.getColumns().stream()
+                .filter(column -> !column.isHidden())
+                .map(ColumnMetadata::getName)
+                .collect(toImmutableSet());
+
+        Set<String> analyzeColumnNames = getColumnNames(analyzeProperties)
+                .map(columnNames -> {
+                    // validate that proper column names are passed via `columns` analyze property
+                    if (columnNames.isEmpty()) {
+                        throw new TrinoException(INVALID_ANALYZE_PROPERTY, "Cannot specify empty list of columns for analysis");
+                    }
+                    if (!allColumnNames.containsAll(columnNames)) {
+                        throw new TrinoException(
+                                INVALID_ANALYZE_PROPERTY,
+                                format("Invalid columns specified for analysis: %s", Sets.difference(columnNames, allColumnNames)));
+                    }
+                    return columnNames;
+                })
+                .orElse(allColumnNames);
+
+        return new IcebergAnalyzeHandle(
+                handle.getSchemaName(),
+                handle.getTableName(),
+                handle.getSnapshotId(),
+                handle.getTableSchemaJson(),
+                handle.getPartitionSpecJson(),
+                handle.getFormatVersion(),
+                handle.getNameMappingJson(),
+                handle.getTableLocation(),
+                handle.getStorageProperties(),
+                analyzeColumnNames);
+    }
+
+    @Override
+    public TableStatisticsMetadata getStatisticsCollectionMetadata(ConnectorSession session, ConnectorTableMetadata tableMetadata)
+    {
+        Optional<Set<String>> analyzeColumns = getAnalyzeColumns(tableMetadata.getProperties());
+        Set<ColumnStatisticMetadata> columnStatistics = tableMetadata.getColumns().stream()
+                .filter(column -> !column.isHidden())
+                .filter(column -> analyzeColumns.isEmpty() || analyzeColumns.get().contains(column.getName()))
+                // TODO: add support for NDV summary/sketch, but using Theta sketch, not HLL; see https://github.com/apache/iceberg-docs/pull/69
+                .map(column -> new ColumnStatisticMetadata(column.getName(), NUMBER_OF_DISTINCT_VALUES))
+                .collect(toImmutableSet());
+        return new TableStatisticsMetadata(columnStatistics, ImmutableSet.of(), ImmutableList.of());
+    }
+
+    @Override
+    public ConnectorTableHandle beginStatisticsCollection(ConnectorSession session, ConnectorTableHandle tableHandle)
+    {
+        IcebergAnalyzeHandle handle = (IcebergAnalyzeHandle) tableHandle;
+        if (handle.getFormatVersion() < 2) {
+            throw new TrinoException(NOT_SUPPORTED, "Iceberg table statistics updates require at least format version 2");
```

Comment:
How is merging the spec change going to work? Does that get patched into v2 or does it get included in spec v3?

Reply:
it may get applied to v1 as well. not sure